Source/WebKit/WebProcess/WebPage/WebCookieJar.cpp

```diff
@@ -177,14 +178,26 @@ void WebCookieJar::setCookies(WebCore::Document& document, const URL& url, const
     WebProcess::singleton().ensureNetworkProcessConnection().connection().send(Messages::NetworkConnectionToWebProcess::SetCookiesFromDOM(document.firstPartyForCookies(), sameSiteInfo, url, frameID, pageID, applyTrackingPreventionInNetworkProcess, cookieString, shouldRelaxThirdPartyCookieBlocking(webFrame)), 0);
 }
 
-void WebCookieJar::cookiesAdded(const String& host, const Vector<WebCore::Cookie>& cookies)
+void WebCookieJar::cookiesAdded(const String& host, Vector<WebCore::Cookie>&& cookies)
 {
-    m_cache.cookiesAdded(host, cookies);
+    auto it = m_changeListeners.find(host);
+    if (it == m_changeListeners.end())
+        return;
+
+    auto cookieChangeListenersHashSet = it->value;
+    for (auto& cookieChangeListener : cookieChangeListenersHashSet)
+        cookieChangeListener.cookiesAdded(host, cookies);
 }
 
-void WebCookieJar::cookiesDeleted(const String& host, const Vector<WebCore::Cookie>& cookies)
+void WebCookieJar::cookiesDeleted(const String& host, Vector<WebCore::Cookie>&& cookies)
 {
-    m_cache.cookiesDeleted(host, cookies);
+    auto it = m_changeListeners.find(host);
+    if (it == m_changeListeners.end())
+        return;
+
+    auto cookieChangeListenersHashSet = it->value;
```

Comment:
Instead of doing your own WeakHashSet copy here, you could also just call WeakHashSet::forEach() which iterates safely on a copy for you:
```
it->value.forEach([&](auto& listener) {
    listener.cookiesDeleted(host, cookies);
});
```

Also, I think we can just use `listener` instead of the very long `cookieChangeListener`.